proto/internal/temporal/server/api/enums/v1/common.proto

```diff
@@ -47,6 +47,42 @@ enum SchedulerState {
     SCHEDULER_STATE_EXECUTING = 2;
 }
 
+enum Scheduler2State {
+    // Default value, unspecified state.
+    SCHEDULER2_STATE_UNSPECIFIED = 0;
+    // Scheduler's top-level state machine has been initialized and is running.
+    SCHEDULER2_STATE_RUNNING = 1;
+}
+
+// State for the state machine scheduler's Generator.
+enum SchedulerGeneratorState {
+    // Default value, unspecified state.
+    SCHEDULER_GENERATOR_STATE_UNSPECIFIED = 0;
+    // Generator will alternate between buffer actions and sleeping for a wakeup
+    // interval.
+    SCHEDULER_GENERATOR_STATE_BUFFERING = 1;
+}
+
+// State for the state machine scheduler's Executor.
+enum SchedulerExecutorState {
+    // Default value, unspecified state.
+    SCHEDULER_EXECUTOR_STATE_UNSPECIFIED = 0;
+    // Executor is awaiting actions to be buffered and eligible for execution.
+    SCHEDULER_EXECUTOR_STATE_WAITING = 1;
+    // Executor is starting actions.
+    SCHEDULER_EXECUTOR_STATE_EXECUTING = 2;
+}
+
+// State for the state machine scheduler's Backfiller.
+enum SchedulerBackfillerState {
```

Comment:
Hmm... I thought the backfiller is just another type of generator that sends requests to the executor.